src/list.rs

```diff
@@ -32,8 +33,47 @@ fn handle_list(app_state: &mut AppState, stdout: &mut StdoutLock) -> Result<()>
 
                 list_state.message.clear();
 
+                let curr_key = key.code;
+
+                if is_searching {
+                    match curr_key {
+                        KeyCode::Esc | KeyCode::Enter => {
+                            is_searching = false;
+                            list_state.search_query.clear();
+                            continue;
+                        }
+                        KeyCode::Char(k) => {
+                            list_state.search_query.push(k);
+                            list_state.message.push_str("search:");
+                            list_state.message.push_str(&list_state.search_query);
+                            list_state.message.push('|');
+
+                            list_state.select_if_matches_search_query();
+
+                            list_state.draw(stdout)?;
+                            continue;
+                        }
+                        KeyCode::Backspace => {
+                            list_state.search_query.pop();
+                            list_state.message.push_str("search:");
+                            list_state.message.push_str(&list_state.search_query);
+                            list_state.message.push('|');
+
+                            list_state.select_if_matches_search_query();
+
+                            list_state.draw(stdout)?;
+                            continue;
+                        }
+                        _ => {
+                            continue;
+                        }
+                    }
+                }
+
                 match key.code {
-                    KeyCode::Char('q') => return Ok(()),
```

Comment:
Why?

Reply:
This already existed in the source code, [ref](https://github.com/rust-lang/rustlings/blob/main/src/list.rs#L36), if you are asking why I made it NOT a single line and added the `{` then those are remnants from me experimenting around, will fic it